layers/meta-resin-raspberrypi/recipes-support/hostapp-update-hooks/files/999-resin-boot-cleaner

```diff
@@ -0,0 +1,27 @@
+#!/bin/sh
+
+#
+# Script which configures the cleans up redundant files in the boot partition
+#
+
+set -o errexit
+
+. /usr/sbin/resin-vars
+
+DURING_UPDATE=${DURING_UPDATE:-0}
+
+if [ "$DURING_UPDATE" = "1" ]; then
+	if  grep -q "ext4load" "$RESIN_BOOT_MOUNTPOINT/boot.scr" ; then
+		if [ -e "$RESIN_BOOT_MOUNTPOINT/zImage" ]; then
+			# Testing the destination sysroot kernel is tricky. Lets check current root
+			# instead as all future roots will have the kernel in root partition in /boot
+			# This leaves the kernel lying around in boot partition until the next HUP
+			if [ -e "/boot/zImage" ]; then
+				printf "[INFO] u-boot boot.scr now reads kernel from root partitions but found zImage in boot \n"
+				printf "[INFO] removing $RESIN_BOOT_MOUNTPOINT/zImage \n"
```

Comment:
Why do we have this script rpi specific? I'd see this as a generic cleaner hook which would handle various migrations.

Reply:
Got it.